### PR TITLE
Google Authentication for Vendor 2/2 : Frontend Interface 

### LINKED ISSUE #16: feature : Google OAuth 2.0 integration for Vendor Sign In
**Is your feature request related to a problem? Please describe.**

I want to implement Google OAuth, which will allow vendor to sign in not only with usual email and password but also using Google.

**Describe the solution you'd like**

I would use the Google OAuth 2.0 npm library and Passport JS to implement this feature. There will also be cookies to store session data of user who has logged in and 401 failure pages in case login does not succeed. Once login is successful, vendor will be able to see the Order List Page. I will implement this redirection route once #9 is resolved.

**Describe alternatives you've considered**

For now I haven't considered any alternatives but Firebase would be another possible solution.
**Additional context**
`NA`

**I would like to be assigned to the issue once the others assigned to me are closed**

### PR DESCRIPTION
**Pull request by vedanthv**
#### Issue Number
fixes #16 

#### Describe the changes you've made
I have integrated the backend Google Authentication configuration to the Frontend webpage.

#### Describe if there is any unusual behaviour of your code ( Write `NA` if there isn't )
After successfully logging and redirecting to the Vendors Orders-List Page, the styling for the page isn't there. Its a plain HTML5 page. I am working on this and will resolve it in the next issue.  Since the codebase for this PR is long, I decided to forward that to the next issue.

#### Additional context
I am not putting any screenshots or demo as I don't want my email id and secret API keys to be exposed during the demo.

#### Test plan ( Optional but Recommended )
- Fetch and merge this code to your fork.
- Get API Keys from Google Dev Console.
- Go to `Vendor Sign-In` in VSCode and spin up node instance using `node index.js`
- Open the main.html file page in your browser.
- Select Sign Up with Google at the bottom of the screen
- Put in your credentials.
- After successful sign in you will be redirected to the Vendors Order List Page.

#### Checklist

- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: PRNTRS Vendor/Vendor Sign-In/views/index.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html>
+<html lang="en" dir="ltr">
+
+<head>
+
+    <meta charset="utf-8">
+    <title>Vendor Sign In</title>
+    <link rel="stylesheet" href="styles.css">
```
Comment (A-HK) [inline PRNTRS Vendor/Vendor Sign-In/views/index.html]: The `href` links to `styles.css` but the `styles.css` file is moved to the `public` directory from what I can see here, so shouldn't the path be different?
What am I missing here @vedanthv ?

Comment (vedanthv) [reply PRNTRS Vendor/Vendor Sign-In/views/index.html]: @A-HK You can read why I have done that in the code review here: https://github.com/Tech-Matrix/PRNTRS/pull/28

**Comment by A-HK** [comment]
Can you actually please send a screenshot of the changed vendor login now @vedanthv ?
I understand about the keys and stuff, I don't need a demo, just a pic for some context.
Because strangely, no styling's been applied on any html page when I pull your PR locally.

I notice that you've mentioned in your `next issue` , did you mean that the css doesn't work for all the html pages?

**Comment by vedanthv** [comment]
> I notice that you've mentioned in your `next issue` , did you mean that the css doesn't work for all the html pages?

@A-HK That's right. The CSS isn't working for any pages that have been rendered from the backend. Its mostly because there are so many `style.css` pages, its impossible to render the styling without any conflicting classes with the other pages. I will refactor and rename the CSS stylesheets later. It takes time to check if the classes are not conflicting each other

**Comment by vedanthv** [comment]
@A-HK Also I notice your fork https://github.com/A-HK/PRNTRS is 25 commits behind the main(this original repository). You would need to fetch and pull to notice all changes on your local system!
image

**Comment by A-HK** [comment]
Right now I'm pulling and checking right from the Tech-Matrix owned repo ; not making any changes as such from my end, hence not using my fork.

**Comment by A-HK** [comment]
> Right now I'm pulling and checking right from the Tech-Matrix owned repo ; not making any changes as such from my end, hence not using my fork.

I also do this because it's way easier to use GitHub CLI to check the PRs out, and the PRs come to this repo, not my fork xD.

**Comment by vedanthv** [comment]
> > Right now I'm pulling and checking right from the Tech-Matrix owned repo ; not making any changes as such from my end, hence not using my fork.
> 
> I also do this because it's way easier to use GitHub CLI to check the PRs out, and the PRs come to this repo, not my fork xD.

Hmm I see. I still prefer my Terminal and VSCode Diff Checker. Maybe I'm old school!

**Comment by A-HK** [comment]
> > I notice that you've mentioned in your `next issue` , did you mean that the css doesn't work for all the html pages?
> 
> @A-HK That's right. The CSS isn't working for any pages that have been rendered from the backend. Its mostly because there are so many `style.css` pages, its impossible to render the styling without any conflicting classes with the other pages. I will refactor and rename the CSS stylesheets later. It takes time to check if the classes are not conflicting each other

Hey @vedanthv , maybe make a PR to resolve this issue before doing anything else?
I'm sure you understand why I don't feel comfortable merging this until I see that this can be fixed :)

**Comment by vedanthv** [comment]
> Hey @vedanthv , maybe make a PR to resolve this issue before doing anything else?
> I'm sure you understand why I don't feel comfortable merging this until I see that this can be fixed :)

Done. folder refactor is very time consuming and I won't be doing it until Friday for the payment gateway #34 . You can keep the PR open or merge anything's fine.
Also you can get API keys and check success route styling since its a protected route.

**Comment by A-HK** [comment]
Thanks for this @vedanthv . Will merge this.
CSS still messed up though ( at least on my end ) , I trust that you'll resolve this soon?

**Comment by vedanthv** [comment]
@A-HK Are you referring to the CSS on the Orders-List Page or the Sign Up page of the vendor?
Sign - Up Page : Its working absolutely fine.
Orders Page : You have to get the API keys and configure it. It works from my end.

**Comment by A-HK** [comment]
Hmm this is how it looks for me
image

And why is there both an `indexvendor.html` and `vendorindex.html`? As in, what does one file do differently from the other?

**Comment by vedanthv** [comment]
> Hmm this is how it looks for me
> 
> And why is there both an `indexvendor.html` and `vendorindex.html`? As in, what does one file do differently from the other?

@A-HK 

https://user-images.githubusercontent.com/44313631/135193463-b6ad146b-5b84-45a5-a8b4-9d6f4a15aa6c.mp4

I deleted `indexvendor.html` here #40

**Pull request merged**